src/tests/client.test.js

```diff
@@ -0,0 +1,54 @@
+import jest from 'jest-mock';
+
+import { make_client } from '../client.js';
+
+/**
+ * Test the client
+ */
+
+describe('client', () => {
+
+  it('should return results: mock', async () => {
+
+    // keep a copy of fetch
+    const nodeFetch = fetch;
+
+    const mockResponse = {
+      body: {
+        results: [{
+          hits: [{
+            description: 'test',
+            id: 1,
+            name: 'test',
+            postgres_searchbox_v1_doc: "'coalesc':1,3 'descript':4 'name':2",
+          },],
+        },], query: 'test',
+      }
+    }
+
+    global.fetch = jest.fn(() =>
+      Promise.resolve({
+        ok: true,
+        json: () => Promise.resolve(mockResponse)
+      })
+    );
+
+
+    const client = make_client('http://localhost:3000/api/search');
```

Comment:
What makes sure that this URL is operational?  Will it conflict with integration tests?

Reply:
Good question, it's something I just picked up today, so I could be proven wrong. 

As I replace fetch with a mock, the url doesn't actually matter, because the mock always returns the mock response regardles of url.

After the mock the original fetch is restored. I think I should wrap this in a `try` block and restore the original fetch in a `finally`. To make absolutely sure it's restored.

In `package.json` the jest is started with the `--runInBand` flag so the tests are run in series. This should prevent it from effecting other tests.

Reply:
> As I replace fetch with a mock, the url doesn't actually matter, because the mock always returns the mock response regardles of url.

Then this URL should be deliberately incorrect, to avoid any chance of interference with a running server.

Reply:
This makes absolute sense. I forgot to amend but will do in the next PR